src/Base/StringUtilityInterface.php

```diff
@@ -96,6 +96,26 @@ public function removeSuffix($string, $suffix);
      */
     public function removeWhitespace($string);
 
+    /**
+     * Returns the substring after the first occurrence of $after<br />
+     * and before the first occurrence of $before<br />
+     * For example:<br />
+     * substring('there is no foo without a bar', 'no ', ' without') === 'foo'<br />
+     * substring('there is no foo without a bar', 'baz', ' no') === 'there is'<br />
+     * substring('there is no foo without a bar', 'foo ', 'baz') === 'without a bar'<br />
+     * substring('there is no foo without a bar', 'foobar ', 'baz') === 'there is no foo without a bar'<br />
+     * <br />
+     * If $after is not found, this function will return the string from the beginning.<br />
+     * If $before is not found, this function will return the string until the end.<br />
+     *
+     * @param string $string
+     * @param string $after
+     * @param string $before
+     *
+     * @return string
+     */
+    public function substring($string, $after, $before);
```

Comment:
Ich finde den Namen "substringBetween" passender, da es schon ein "substringBefore" und ein "substringAfter" gibt. Ich werde dir ein entsprechenden Pull-Request zukommen lassen :-).

Mit den Variablennamen "$after" und "$before" bin ich auch sehr unzufrieden. Nach langem probieren waren beide Namen dann das "geringste Uebel".
Gerade sind mir "$startAfter" und "$untilBefore" eingefallen. Die Woerter "Zeichen", "Marker", "Limiter", "Delimiter" schwirren auch noch in meinem Kopf rum, ergeben aber noch keine Bezeichnung mit der ich zufrieden bin.
Es ist in der Tat schwer zu fassen, da die uebergebenen Zeichenketten nicht im Substring enthalten sind (was bei meinen Fallbeispielen aber immer gewuenscht ist).